### PR TITLE
feat: cache messages

### PR DESCRIPTION
**Pull request by carlbrugger**

### CODE REVIEW

**Review by brentkulwicki — APPROVED** [review]

**Review by ashleygmulligan2 — COMMENTED** [review]
Does this tie back to an issue? Can you summarize what the problem was. This will help with changelog.

**Comment by ashleygmulligan2** [comment]
Platform wasn't sending the record messages to the plugin. A platform change was made to enable this, now the plugin is following. Can you link back to the platform change in your PR description, too?
